src/charm.py

```diff
@@ -694,39 +697,148 @@ def _activate_charm(self) -> None:
             self._start_service()
             self.model.unit.status = ActiveStatus()
 
-    def _on_add_admin_user_action(self, event: ActionEvent) -> None:
-        """Add a new admin user to Discourse.
+    def _on_promote_user_action(self, event: ActionEvent) -> None:
+        """Promote a user to a specific trust level.
```

Comment:
The logic of this function is simple, but the function is very long and so it is difficult to see the real logic. I believe splitting it into several functions will help with readability. This will imply moving the code that calls pebble to its own functions.

Reply:
this has been addressed in same way as create user